doc: improve-tutorial-last-steps (#132)
eigerco/pallet-move: doc/tutorial.md

- [Overview](#overview)
- [Setup and Code Example](#setup-and-code-example)
- [Publishing Modules](#publishing-modules)

> - - [Estimating Gas for Module Publication](#estimating-gas-for-module-publication)
> - - [Publication](#publication)

Reviewer: Would this fix the rendering problem?
```suggestion
  - [Estimating Gas for Module Publication](#estimating-gas-for-module-publication)
  - [Publication](#publication)
```

Author: What do you see? :upside_down_face: I did not see any issues...

Author: Done

Estimated gas: Estimate (gas_used: 69, vm_status_code: EXECUTED)
```

> ### Publication 

Reviewer: While I agree this is a more fancy word, I would stay with `Publishing modules`, since even the `aptos` command uses such wording:
```
  publish
          Publishes the modules in a Move package to the Aptos blockchain
```

![Script execution with adjusted __chequeLimit__](assets/polkadot.js_execute_script_buy.png)

## Finishing the Tutorial

> You have successfully published a Move module on your Polkadot blockchain and executed a Move script successfully.
> 
> You can iterate the full demonstration example for training purposes or personal interest. So far, the developer Bob has published his module for his car wash and initialized it by executing the script `initial_coin_minting`. It's time for Alice (Bob's first customer) to register, buy a coin and wash her car! Therefore, try to:
> - create transactions,
> - estimate the optimal amount of needed gas,
> - and execute them as user `Alice` for the following scripts:
>   - `register_new_user`,
>   - `buy_coin` (observe here Bob's and Alice's account balances, because the price of a washing coin is 1 UNIT),
>   - `wash_car`.

Reviewer: ```suggestion
You have successfully published a Move module on your Polkadot blockchain and executed a Move script successfully.

You can iterate the full demonstration example for training purposes or personal interest. So far, the developer Bob has published his module for his car wash and initialized it by executing the script `initial_coin_minting`. It's time for Alice (Bob's first customer) to register, buy a coin and wash her car! Therefore, try to:
- create transactions,
- estimate the optimal amount of needed gas,
- and execute them as user `Alice` for the following scripts:
  - `register_new_user`,
  - `buy_coin` (observe here Bob's and Alice's account balances, because the price of a washing coin is 1 UNIT),
  - `wash_car`.
```

> You can iterate the full demonstration example for training purposes or personal interest. So far, the developer Bob has published his module for his car wash and initialized it by executing the script `initial_coin_minting`. It's time for Alice (Bob's first customer) to register, buy a coin and wash her car! Therefore, try to:
> - create transactions,
> - estimate the optimal amount of needed gas,
> - and execute them as user `Alice` for the following scripts:
>   - `register_new_user`,
>   - `buy_coin` (observe here Bob's and Alice's account balances, because the price of a washing coin is 1 UNIT),
>   - `wash_car`.

Reviewer: ```suggestion
So now we are ready to buy some coins for washing our car using our Substrate balance.

First, let's check the balance before we execute some new scripts:

<@micha: insert here a balance pallet RPC for getting Alice's funds oooor use a screenshot instead>

Now, let's execute the following actions:

1. Using Alice's account, let's register her as a customer by executing the `register_new_user.mv` script
 - Hint 1: this script requires one signer which has to be Alice's account ID.
 - Hint 2: use the estimation RPC method to calculate gas.
 - Hint 3: `cheque_amount` is unused in this script, so set it to zero.
 
2. Now, again using Alice's account, buy a coin with a `buy_coin.mv` script:
  - Hint 1: this script requires one signer which has to be Alice's account ID.
  - Hint 2: use the estimation RPC method to calculate gas.
  - Hint 1: because the price of a washing coin is 1 UNIT), we need to write a cheque of around xxxx
 
3. Wash the car with Alice's account 
  - Hint 1: ...

After the car is washed, now check Alice's balance again and see how the above operations changed her balance status:

  <@micha: insert here a balance pallet RPC for getting Alice's funds oooor use a screenshot instead>
```

Author: Done

![Script execution with adjusted __chequeLimit__](assets/polkadot.js_execute_script_buy.png)

## Finishing the Tutorial

> You have successfully published a Move module on your Polkadot blockchain and executed a Move script successfully. The past sections have explained how to:
> - [estimate the optimal amount of gas for publishing a module](#estimating-gas-for-module-publication),
> - [publish a module](#publication),
> - [create the script transaction](#creating-script-transactions),
> - [estimate the optimal amount of gas for executing a script](#estimating-gas-for-script-execution),
> - and how to [execute a script](#execution).

Reviewer: Let's keep it short in order not to scare away people with a long tutorial page.
```suggestion
You have successfully published a Move module on your Polkadot blockchain and executed a Move script successfully.
```

Author: Done
